ui: remove custom charts extra padding

vladlos opened the pull request:
before: 900px padding was added to provide space for chart
legend in the end of metrics page, but this rule corrupt a view
of custom chart page

after: made this padding specific to metric page, calculate it
dynamicly based on node number, this removed mentioned
issue with custom chart page.

Resolves: #64791

Release note(ui): fix custom charts page layout

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/66261)
<!-- Reviewable:end -->

koorosh reviewed (commented):
As I understand, this particular fix is required when graph should display a lot of series and this is a case when cluster has lots of nodes.
I would suggest to apply this "margin" fix in case when cluster has more than X nodes so this increased margin won't break page layout for small clusters.
Just in case, chart displays legend as floating tooltip if cluster has less than 10 nodes.

> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66261#-:-MbqUQhh2d9svRp-5GKs:b-g0ev0b)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @nathanstilwell)



<!-- Sent from Reviewable.io -->

vladlos reviewed (commented):
@koorosh updated to calculate padding based on nodes ammount:
![Screenshot 2021-06-14 at 16.36.00.png](https://files.reviewableusercontent.io/images/reviewable/12850886/pT8lQ7qAELUnyJu7YxlUnV5X/Screenshot%202021-06-14%20at%2016.36.00.png)
![Screenshot 2021-06-14 at 16.34.48.png](https://files.reviewableusercontent.io/images/reviewable/12850886/tzYElLYC1tKJxS8BfC1ZqRNJ/Screenshot%202021-06-14%20at%2016.34.48.png)
![Screenshot 2021-06-14 at 16.24.46.png](https://files.reviewableusercontent.io/images/reviewable/12850886/VKXVUXlTRokAUlb8AAuaMxr7/Screenshot%202021-06-14%20at%2016.24.46.png)

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66261#-:-Mc9kjN5-3uyzhwZmebx:b-ultih9)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @koorosh and @nathanstilwell)



<!-- Sent from Reviewable.io -->

nathanstilwell reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66261)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @elkmaster and @koorosh)

___
*[pkg/ui/src/views/cluster/containers/nodeGraphs/index.tsx, line 248 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/66261#-McF1QEN5QF5_kGJ-Vu6:-McF1QEO2F26-nOKpCR0:b-hj9bgz) ([raw file](https://github.com/cockroachdb/cockroach/blob/ea563f7419080702511886b44c2e023c67f4513c/pkg/ui/src/views/cluster/containers/nodeGraphs/index.tsx#L248)):*
> ```XML
>     });
> 
>     // add pading to have last chart tooltip visible
> ```

I am into dynamically determining the size of the space for the legend, but could I see some additional comments here describing why this particular algorithm? Or just an explanation of what this is doing? Even if it is something as simple as "When there are more than 8 node IDs we divide the number by three, for a three column layout assuming 10px for each element and add an additional 90px. Otherwise set the padding to 50px".


<!-- Sent from Reviewable.io -->

vladlos reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66261)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @koorosh and @nathanstilwell)

___
*[pkg/ui/src/views/cluster/containers/nodeGraphs/index.tsx, line 248 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/66261#-McF1QEN5QF5_kGJ-Vu6:-McJrKkU1GFIB4mLu6U8:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/ea563f7419080702511886b44c2e023c67f4513c/pkg/ui/src/views/cluster/containers/nodeGraphs/index.tsx#L248)):*
<details><summary><i>Previously, nathanstilwell (Nathan Stilwell) wrote…</i></summary><blockquote>

I am into dynamically determining the size of the space for the legend, but could I see some additional comments here describing why this particular algorithm? Or just an explanation of what this is doing? Even if it is something as simple as "When there are more than 8 node IDs we divide the number by three, for a three column layout assuming 10px for each element and add an additional 90px. Otherwise set the padding to 50px".
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

Santamaura reviewed (approved):
> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66261)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @koorosh and @nathanstilwell)



<!-- Sent from Reviewable.io -->

Santamaura reviewed (approved):
> Reviewed 1 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66261)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @koorosh and @nathanstilwell)



<!-- Sent from Reviewable.io -->

Santamaura commented:
Code change with the new comments and screenshots of updated visuals LGTM

koorosh reviewed (approved):
LGTM

nathanstilwell reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66261)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @koorosh)



<!-- Sent from Reviewable.io -->

nathanstilwell commented:
bors cancel

vladlos committed:
ui: remove custom charts extra padding

before: 900px padding was added to provide space for chart
legend in the end of metrics page, but this rule corrupt a view
of custom chart page

after: made this padding specific to metric page, calculate it
dynamicly based on node number, this removed mentioned
issue with custom chart page.

Resolves: cockroachdb#64791

Release note(ui): fix custom charts page layout

elkmaster commented:
bors r+
